### PR TITLE
Add minimal-git-server

### PR DESCRIPTION
**Pull request by mcarbonne**
Original PR :  #628, software is now old enough (initial version released in January 2024).

To ensure your Pull Request is dealt with swiftly, please check the following (check the boxes `[x]`):
- [x] Submit one item per pull request. This eases reviewing and speeds up inclusion.
- [x] You have searched the repository for any relevant [issues](https://github.com/awesome-selfhosted/awesome-selfhosted-data/issues) or [PRs](https://github.com/awesome-selfhosted/awesome-selfhosted-data/pulls), including closed ones.
- [x] Any software you are adding is not already listed at any of [awesome-sysadmin](https://github.com/n1trux/awesome-sysadmin), [staticgen.com](https://www.staticgen.com/), [staticsitegenerators.net](https://staticsitegenerators.net/), [dbdb.io](https://dbdb.io/browse).
- [x] The file you are adding is formatted as described in [addition.md](https://github.com/awesome-selfhosted/awesome-selfhosted-data/blob/master/.github/ISSUE_TEMPLATE/addition.md).
- [x] `Demo` links should only be used for interactive demos, i.e. not video demonstrations. 
- [x] Comments and unused optional fields have been removed.
- [x] The file you are adding uses [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) file naming, for example `my-awesome-software.yml`.
- [x] Values for `platform` are the main **server-side** requirements for the software. Don't include frameworks or specific dialects.
- [x] Any software project you are adding to the list is actively maintained.
- [x] Any software project you are adding was first released more than 4 months ago.
- [x] Any software project you are adding has working installation instructions.
- [x] You understand that your Pull Request will be merged ~1 week after approval, to allow for further comments if needed.

### CODE REVIEW

File: software/minimal-git-server.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: minimal-git-server
+website_url: https://github.com/mcarbonne/minimal-git-server
+source_code_url: https://github.com/mcarbonne/minimal-git-server
+description: A lightweight and minimal self-hosted git server with a basic CLI to manage repositories, supporting multiple accounts and running in a container.
+licenses:
+  - MIT
+platforms:
+  - Shell
```
[the remark this answers is not in the document]
Comment (mcarbonne) [reply software/minimal-git-server.yml:8]: ```Values for platform are the main server-side requirements for the software. Don't include frameworks or specific dialects.```
I don't get it: [minimal-git-shell](https://github.com/mcarbonne/minimal-git-server) is written (mostly) in bash/shell scripts, isn't it OK ?

[the remark this answers is not in the document]
Comment (nodiscc) [reply software/minimal-git-server.yml:8]: @mcarbonne it mostly depends on the installation method/requirements. Where can I find installation instructions? I can't find them in the README

[the remark this answers is not in the document]
Comment (mcarbonne) [reply software/minimal-git-server.yml:8]: Everything runs inside a container and instructions are available here: https://github.com/mcarbonne/minimal-git-server?tab=readme-ov-file#minimal-configuration (a simple docker run).

[the remark this answers is not in the document]
Comment (kokomo123) [reply software/minimal-git-server.yml:8]: Seems simple enough for me, so I think I can approve this PR (pending other thoughts from collaborators)

**Review by kokomo123 — APPROVED** [review]
LGTM, 
EDIT: I approved this while the changes still need to be made. oops

File: software/minimal-git-server.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nodiscc) [reply software/minimal-git-server.yml:8]: If Docker is the only supported installation method, the `platforms` list must only contain `Docker`

**Review by nodiscc — APPROVED** [review]
Thanks

**Comment by nodiscc** [comment]
@kokomo123 don't worry about missing some things in the review, it is still helpful. Thank you for your help

**Pull request merged**